src/middleware/aws/operation/dynamodb.rs

```diff
@@ -0,0 +1,143 @@
+use crate::{semcov, StringValue};
+
+use super::*;
+
+aws_target!(DynamoDBOperation);
```

Comment:
It could even be a real empty type: [Empty Types](https://doc.rust-lang.org/nomicon/exotic-sizes.html#empty-types)

Reply:
On the other hand, with a custom struct we could define sdk-specific methods, i.e.:

```rust
impl<'a> DynamoDBOperation<'a> {
    #[inline]
    pub fn index_name(self, index_name: impl Into<StringValue>) -> AwsSpan {
        self.0.attribute(semcov::AWS_DYNAMODB_INDEX_NAME.string(index_name))
    }
}
```

but I don't think we need any

Reply:
I tried the solution with [Empty Types](https://doc.rust-lang.org/nomicon/exotic-sizes.html#empty-types) instead of proxy structs and it actually looks better for me:

https://github.com/nentgroup/telemetry-rust/pull/16

Reply:
Approved #16 and it does look neater!